src/shims/posix/fs.rs

```diff
@@ -251,22 +251,70 @@ impl FileDescriptor for io::Stderr {
     }
 }
 
+#[derive(Debug)]
+struct DevNull;
+
+impl FileDescriptor for DevNull {
+    fn as_file_handle<'tcx>(&self) -> InterpResult<'tcx, &FileHandle> {
+        throw_unsup_format!("/dev/null cannot be used as FileHandle");
+    }
+
+    fn read<'tcx>(
+        &mut self,
+        _communicate_allowed: bool,
+        _bytes: &mut [u8],
+    ) -> InterpResult<'tcx, io::Result<usize>> {
+        throw_unsup_format!("cannot read from /dev/null");
+    }
+
+    fn write<'tcx>(
+        &self,
+        _communicate_allowed: bool,
+        bytes: &[u8],
+    ) -> InterpResult<'tcx, io::Result<usize>> {
+        // We just don't write anything
+        Ok(Ok(bytes.len()))
+    }
+
+    fn seek<'tcx>(
+        &mut self,
+        _communicate_allowed: bool,
+        _offset: SeekFrom,
+    ) -> InterpResult<'tcx, io::Result<u64>> {
+        throw_unsup_format!("cannot seek on /dev/null");
+    }
+
+    fn close<'tcx>(
+        self: Box<Self>,
+        _communicate_allowed: bool,
+    ) -> InterpResult<'tcx, io::Result<i32>> {
+        throw_unsup_format!("/dev/null cannot be closed");
+    }
+
+    fn dup<'tcx>(&mut self) -> io::Result<Box<dyn FileDescriptor>> {
+        Ok(Box::new(DevNull))
+    }
+}
+
 #[derive(Debug)]
 pub struct FileHandler {
     handles: BTreeMap<i32, Box<dyn FileDescriptor>>,
 }
 
-impl<'tcx> Default for FileHandler {
-    fn default() -> Self {
+impl<'tcx> FileHandler {
+    pub(crate) fn new(drop_stdout_stderr: bool) -> FileHandler {
         let mut handles: BTreeMap<_, Box<dyn FileDescriptor>> = BTreeMap::new();
-        handles.insert(0i32, Box::new(io::stdin()));
-        handles.insert(1i32, Box::new(io::stdout()));
+        if drop_stdout_stderr {
+            handles.insert(0i32, Box::new(DevNull));
+            handles.insert(1i32, Box::new(DevNull));
+        } else {
+            handles.insert(0i32, Box::new(io::stdin()));
+            handles.insert(1i32, Box::new(io::stdout()));
+        }
         handles.insert(2i32, Box::new(io::stderr()));
         FileHandler { handles }
```

Comment:
cc https://github.com/rust-lang/miri/issues/2143

@oli-obk perhaps the problem is that here, you touched stdout and STDIN but the handling of STDERR (line 264/314) actually remains unchanged? :sweat_smile:

Reply:
Aaaaa dammit